src/NHibernate/Dialect/AbstractHanaDialect.cs

```diff
@@ -0,0 +1,1003 @@
+		{
+			return new HanaDataBaseSchema(connection);
+		}
+
+		#endregion
+
+		#region Lock acquisition support
+
+		/// <inheritdoc />
+		[Obsolete("Use UsesColumnsWithForUpdateOf instead")]
+		public override bool ForUpdateOfColumns => true;
+
+		/// <inheritdoc />
+		public override string GetForUpdateString(string aliases)
+		{
+			return ForUpdateString + " of " + aliases;
+		}
+
+		/// <inheritdoc />
+		public override string ForUpdateNowaitString
+		{
+			get { return ForUpdateString + " nowait"; }
+		}
+
+		/// <inheritdoc />
+		public override string GetForUpdateNowaitString(string aliases)
+		{
+			return GetForUpdateString(aliases) + " nowait";
+		}
+
+		#endregion
+
+		#region Table support
+
+		#region Temporary table support
+
+		/// <inheritdoc />
+		public override bool SupportsTemporaryTables => true;
+
+		/// <inheritdoc />
+		public override string GenerateTemporaryTableName(string baseTableName)
+		{
+			return "#HT_" + baseTableName;
+		}
+
+		/// <inheritdoc />
+		public override bool? PerformTemporaryTableDDLInIsolation()
+		{
+			return false;
+		}
+
+		/// <inheritdoc />
+		public override bool DropTemporaryTableAfterUse()
+		{
+			return true;
+		}
+
+		#endregion
+
+		#endregion
+
+		#region Callable statement support
+
+		/// <inheritdoc />
+		public override int RegisterResultSetOutParameter(DbCommand statement, int position)
+		{
+			// Result set (TABLE) OUT parameters don't need to be registered
+			return position;
+		}
+
+		#endregion
+
+		#region Current timestamp support
+
+		/// <inheritdoc />
+		public override bool SupportsCurrentTimestampSelection => true;
+
+		/// <inheritdoc />
+		public override long TimestampResolutionInTicks
+		{
+			get { return 10L; } // Maximum precision (one tick)
+		}
+
+		#endregion
+
+		#region Constraint support
+
+
+		/// <inheritdoc />
+		public override string CascadeConstraintsString
+		{
+			get { return " cascade"; }
+		}
+
+		#endregion
+
+		#region Native identifier generation
+
+		#region IDENTITY support
+
+		/// <inheritdoc />
+		public override bool SupportsIdentityColumns => true;
+
+		/// <inheritdoc />
+		public override string GetIdentitySelectString(string identityColumn, string tableName, DbType type)
+		{
+			return IdentitySelectString + tableName;
+		}
+
+		/// <inheritdoc />
+		public override string IdentitySelectString
+		{
+			get { return "select current_identity_value() from "; }
+		}
+
+		/// <inheritdoc />
+		public override string IdentityColumnString
+		{
+			get { return "generated by default as identity"; }
+		}
+
+		#endregion
+
+		#region SEQUENCE support
+
+		/// <inheritdoc />
+		public override bool SupportsSequences => true;
+
+		/// <inheritdoc />
+		public override bool SupportsPooledSequences => true;
+
+		/// <inheritdoc />
+		public override string GetSequenceNextValString(string sequenceName)
+		{
+			return "select " + GetSelectSequenceNextValString(sequenceName) + " from dummy";
```

Comment:
The `DUMMY` table is always available in HANA either directly as `SYS.DUMMY` or via the public synonym `PUBLIC.DUMMY`. While it is generally true that another `DUMMY` table can cause issues, it's not an issue in this specific case because the query is not actually selecting anything from the table. The value returned from the statement is always the next value from the sequence. The `FROM` clause is only required to complete the SQL statement. In fact, you could use any other table as well, but since a table named `DUMMY` is always available, I'm using this one.

Reply:
> it's not an issue in this specific case

Add an empty dummy table, and try selecting nextval from it: it will yield `null`. So it can be an issue for this case too.

Reply:
Looks like you're right. It should be changed to `sys.dummy`.